### ISSUE TITLE
Get empty File

### ISSUE THREAD

**Issue by JNKielmann** [minio/minio-java#636]
Hi,

I uploaded an empty file (length = 0 Bytes) to minio.
If I try to download it with the getObject Method I get the following error:
```
ErrorResponse(code=InvalidRange, message=The requested range is not satisfiable, bucketName=null, objectName=null, resource=/my-bucket/test, requestId=3L137, hostId=3L137)
request={method=GET, url=http://127.0.0.1:9000/my-bucket/test, headers=Range: bytes=0-
Host: 127.0.0.1:9000
```
I have created a minimal example that reproduces the issue here:
https://gist.github.com/JNKielmann/cd7283b0d7ec9cc57b7fea36c46395b0

The problem seems to be the Range header: `bytes=0-` because the file has no bytes. If you remove this entry from the request it succeeds.

Is this a bug or are empty files just not supported? If it's a bug I can make a PR if you want.

**Comment by harshavardhana** [comment]
Looks like this is a bug that occurs with AWS S3 as well. 

```
Test: getObject(String bucketName, String objectName)
error occured
ErrorResponse(code=InvalidRange, message=The requested range is not satisfiable, bucketName=null, objectName=null, resource=null, requestId=D853EE1C55FB9B3D, hostId=0nB1RJAKXtpgTFy5z33/W9ucRPS+qfeUCkWmz63uhgcWTd/MnMaI7NW/NwUxDkjdRVEZ43eAr9I=)
request={method=GET, url=https://minio-java-test-28g69d9.s3.amazonaws.com/minio-java-test-1fpbcbt, headers=Range: bytes=0-
Host: minio-java-test-28g69d9.s3.amazonaws.com
User-Agent: Minio (amd64; amd64) minio-java/dev
x-amz-content-sha256: UNSIGNED-PAYLOAD
x-amz-date: 20171030T073441Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20171030/us-east-1/s3/aws4_request, SignedHeaders=host;range;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=416, headers=x-amz-request-id: D853EE1C55FB9B3D
x-amz-id-2: 0nB1RJAKXtpgTFy5z33/W9ucRPS+qfeUCkWmz63uhgcWTd/MnMaI7NW/NwUxDkjdRVEZ43eAr9I=
Content-Type: application/xml
Transfer-Encoding: chunked
Date: Mon, 30 Oct 2017 07:34:43 GMT
Server: AmazonS3
}

        at io.minio.MinioClient.execute(MinioClient.java:1201)
        at io.minio.MinioClient.executeGet(MinioClient.java:1294)
        at io.minio.MinioClient.getObject(MinioClient.java:1592)
        at io.minio.MinioClient.getObject(MinioClient.java:1506)
        at FunctionalTest.getObject_test8(FunctionalTest.java:1007)
        at FunctionalTest.runTests(FunctionalTest.java:2475)
        at FunctionalTest.main(FunctionalTest.java:2603)
:functional:runFunctionalTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':functional:runFunctionalTest'.
> Process 'command '/usr/lib/jvm/java-8-openjdk-amd64/bin/java'' finished with non-zero exit value 255

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

```

**Comment by balamurugana** [comment]
@JNKielmann PR https://github.com/minio/minio-java/pull/638 fixes the issue. Please verify.

**Comment by germhaine** [comment]
This bug is present in minio java SDK 3.0.7 .

**Comment by kannappanr** [comment]
The fix is available from releases 3.10 onwards
